framework/db/pgsql/QueryBuilder.php

```diff
@@ -187,16 +187,18 @@ public function resetSequence($tableName, $value = null)
      */
     public function checkIntegrity($check = true, $schema = '', $table = '')
     {
-        $enable = $check ? 'ENABLE' : 'DISABLE';
-        $schema = $schema ? $schema : $this->db->getSchema()->defaultSchema;
-        $tableNames = $table ? [$table] : $this->db->getSchema()->getTableNames($schema);
-        $viewNames = $this->db->getSchema()->getViewNames($schema);
-        $tableNames = array_diff($tableNames, $viewNames);
-        $command = '';
-
-        foreach ($tableNames as $tableName) {
-            $tableName = '"' . $schema . '"."' . $tableName . '"';
-            $command .= "ALTER TABLE $tableName $enable TRIGGER ALL; ";
+        if (empty($table)) {
+            $enable = $check ? 'IMMEDIATE' : 'DEFERRED';
+            $command = "SET CONSTRAINTS ALL $enable; ";
+        } else {
+            $enable = $check ? 'ENABLE' : 'DISABLE';
+            $schema = $schema ? $schema : $this->db->getSchema()->defaultSchema;
+            $viewNames = $this->db->getSchema()->getViewNames($schema);
```

Comment:
I'm using `$viewNames` like previously so the method never tries to disable triggers in a view by accident. Do you think we shouldn't do this test?